pymc3/distributions/distribution.py

```diff
@@ -176,7 +176,23 @@ def __init__(self, shape=(), dtype=None, defaults=('median', 'mean', 'mode'),
 
 
 class DensityDist(Distribution):
-    """Distribution based on a given log density function."""
+    """Distribution based on a given log density function.
+        
+        A distribution with the passed log density function is created, 
+        requires a custom random function passed as kwarg `random` to
+        enable sampling.
+
+        Example:
+        --------
+        .. code-block:: python
+            with pm.Model():
+                normal_dist = pm.Normal.dist()
+                density_dist = pm.DensityDist('density_dist', normal_dist.logp, observed=np.random.randn(100),
+                random=normal_dist.random)
+                step = pm.Metropolis()
```

Comment:
I think this will automatically be assigned `Metropolis`, which might make the code easier to read.  

I'd love if it looked something like 
```
my_rv = scipy.stats.norm(loc=0, scale=1)

with pm.Model():
    density = pm.DensityDist('density', my_rv.logpdf, random=my_rv.rvs)
    trace = pm.sample()
```

but there's something wrong with scipy distributions so that this does not work (so I am just thinking out loud here, and please ignore this).

Reply:
Also, I'm not on the branch, but I think if you pass `observed` to this distribution, you can't sample from the model (because there are no free variables)

Reply:
For the scipy distributions problem you can take a look at https://github.com/pymc-devs/pymc3/blob/0379ef57b2bcd65c6a2bee9eda13579b361019a4/pymc3/tests/test_distributions_random.py#L692 the test passed so I am assuming it must be correct (?) also I had basically just copied the test to the example so maybe that would need rewriting too if this looks incorrect so perhaps you can take a look at that and then get back regarding some problems in it?

Reply:
I'm starting to become a bit suspicious about the test. Can you make it so that you would expect it to fail and make sure that it does? (E.g. not providing a `random()` function).